File: Kotlin/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.kt
import kotlin.math.sqrt

class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer: Int = 0

        val divisorCount = mutableListOf<Int>()
        divisorCount.add(1)

        for (i in 2 .. number) {
            val sqrt = sqrt(i.toDouble())
            var count = 1

            for (j in 2 .. sqrt.toInt()) {
                if (i % j == 0) {
                    count++
                }
            }

            if (i % sqrt == 0.0) {
                count = count * 2 - 1
            } else {
                count *= 2
            }

            if (count > limit) {
                divisorCount.add(power)
            } else {
                divisorCount.add(count)
            }
        }

        return divisorCount.sumOf { it }
    }
}

File: C#/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cs
using System;

public class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 1;
        
        for(int i = 2; i <= number; i++)
        {
            int sqrt = (int)(Math.Sqrt(i));
            int count = 0;
            for(int j = 1; j <= sqrt; j++)
            {
                if(i % j == 0)
                {
                    count++;
                }
            }
            count = sqrt * sqrt == i ? count * 2 - 1 : count * 2;
            
            answer += count > limit ? power : count;
            
        }
        
        return answer ;
    }
}